letta/orm/sqlalchemy_base.py

```diff
@@ -28,46 +35,68 @@ class SqlalchemyBase(CommonSqlalchemyMetaMixins, Base):
     id: Mapped[str] = mapped_column(String, primary_key=True)
 
     @classmethod
-    def list(
-        cls, *, db_session: "Session", cursor: Optional[str] = None, limit: Optional[int] = 50, **kwargs
-    ) -> List[Type["SqlalchemyBase"]]:
-        """
-        List records with optional cursor (for pagination), limit, and automatic filtering.
+    def get(cls, *, db_session: Session, id: str) -> Optional["SqlalchemyBase"]:
+        """Get a record by ID.
 
         Args:
-            db_session: The database session to use.
-            cursor: Optional ID to start pagination from.
-            limit: Maximum number of records to return.
-            **kwargs: Filters passed as equality conditions or iterable for IN filtering.
+            db_session: SQLAlchemy session
+            id: Record ID to retrieve
 
         Returns:
-            A list of model instances matching the filters.
+            Optional[SqlalchemyBase]: The record if found, None otherwise
         """
-        logger.debug(f"Listing {cls.__name__} with filters {kwargs}")
+        try:
+            return db_session.query(cls).filter(cls.id == id).first()
+        except DBAPIError:
+            return None
+
+    @classmethod
+    def list(
+        cls,
+        *,
+        db_session: "Session",
+        cursor: Optional[str] = None,
+        start_date: Optional[datetime] = None,
+        end_date: Optional[datetime] = None,
+        limit: Optional[int] = 50,
+        query_text: Optional[str] = None,
+        **kwargs,
+    ) -> Union[List[Type["SqlalchemyBase"]], Tuple[Optional[str], List[Type["SqlalchemyBase"]]]]:
```

Comment:
is this because sometime list is paginated, and sometimes now? I think we should either leave a comment, or split up paginated vs. non-paginated `list`